Tiles 3 0 x how to define security role in tiles config file?

VanillaJelly opened the pull request:
reference page : http://tiles.apache.org/framework/tutorial/advanced/security.html
1、in XML definition files:
        <definition name="test.definition"   template="/layout/my-template.jsp>
                 <put-attribute name="header" value="/header.jsp" role="myrole" />
         </definition>
2、in JSP pages, when inserting attributes:
          <tiles:insertAttribute name="header" role="myrole" />
The "header" attribute of the page is blank，How to define the security role？

nlebas committed:
[maven-release-plugin]  copy for branch TILES_3_0_X

git-svn-id: https://svn.apache.org/repos/asf/tiles/framework/branches/TILES_3_0_X@1334240 13f79535-47bb-0310-9956-ffa450edef68

nlebas committed:
[maven-release-plugin] prepare release tiles-parent-3.0.0

git-svn-id: https://svn.apache.org/repos/asf/tiles/framework/branches/TILES_3_0_X@1334246 13f79535-47bb-0310-9956-ffa450edef68

nlebas committed:
[maven-release-plugin] prepare for next development iteration

git-svn-id: https://svn.apache.org/repos/asf/tiles/framework/branches/TILES_3_0_X@1334248 13f79535-47bb-0310-9956-ffa450edef68

nlebas committed:
TILES-549 - Improve code quality (post 3.0 release)
TILES-552: Update the poms and the documentation on the release process
TILES-548: mvn release:perfom fails on Maven 3.0.4

git-svn-id: https://svn.apache.org/repos/asf/tiles/framework/branches/TILES_3_0_X@1360370 13f79535-47bb-0310-9956-ffa450edef68

nlebas committed:
update to autotag 1.1

git-svn-id: https://svn.apache.org/repos/asf/tiles/framework/branches/TILES_3_0_X@1360373 13f79535-47bb-0310-9956-ffa450edef68

nlebas committed:
TILES-548: mvn release:perfom fails on Maven 3.0.4

git-svn-id: https://svn.apache.org/repos/asf/tiles/framework/branches/TILES_3_0_X@1360374 13f79535-47bb-0310-9956-ffa450edef68

nlebas committed:
[maven-release-plugin] prepare release tiles-parent-3.0.1

git-svn-id: https://svn.apache.org/repos/asf/tiles/framework/branches/TILES_3_0_X@1360375 13f79535-47bb-0310-9956-ffa450edef68

nlebas committed:
[maven-release-plugin] prepare for next development iteration

git-svn-id: https://svn.apache.org/repos/asf/tiles/framework/branches/TILES_3_0_X@1360377 13f79535-47bb-0310-9956-ffa450edef68

nlebas committed:
[maven-release-plugin] rollback the release of tiles-parent-3.0.1

git-svn-id: https://svn.apache.org/repos/asf/tiles/framework/branches/TILES_3_0_X@1360430 13f79535-47bb-0310-9956-ffa450edef68

nlebas committed:
TILES-548: mvn release:perfom fails on Maven 3.0.4

git-svn-id: https://svn.apache.org/repos/asf/tiles/framework/branches/TILES_3_0_X@1360431 13f79535-47bb-0310-9956-ffa450edef68

nlebas committed:
[maven-release-plugin] prepare release tiles-parent-3.0.1

git-svn-id: https://svn.apache.org/repos/asf/tiles/framework/branches/TILES_3_0_X@1360434 13f79535-47bb-0310-9956-ffa450edef68

nlebas committed:
[maven-release-plugin] prepare for next development iteration

git-svn-id: https://svn.apache.org/repos/asf/tiles/framework/branches/TILES_3_0_X@1360441 13f79535-47bb-0310-9956-ffa450edef68

michaelsembwever committed:
backport r1427572
 `svn merge -c1427572 ^/tiles/framework/trunk`
 http://svn.apache.org/viewvc?view=revision&revision=1427572


git-svn-id: https://svn.apache.org/repos/asf/tiles/framework/branches/TILES_3_0_X@1427575 13f79535-47bb-0310-9956-ffa450edef68

michaelsembwever committed:
backport doc improvements from Michael Isvy
 `svn merge -c1427704 ^/tiles/framework/trunk`
 http://svn.apache.org/viewvc?view=revision&revision=1427704


git-svn-id: https://svn.apache.org/repos/asf/tiles/framework/branches/TILES_3_0_X@1427705 13f79535-47bb-0310-9956-ffa450edef68

nlebas committed:
TILES-563: Typo on configuration tiles 3 page

git-svn-id: https://svn.apache.org/repos/asf/tiles/framework/branches/TILES_3_0_X@1452632 13f79535-47bb-0310-9956-ffa450edef68

michaelsembwever committed:
have the constant public. 
 there is no way currently to evaluate the fallback in the attribute, only to render it.


git-svn-id: https://svn.apache.org/repos/asf/tiles/framework/branches/TILES_3_0_X@1484307 13f79535-47bb-0310-9956-ffa450edef68

michaelsembwever committed:
back port from trunk r1368433 (TILES-557) and r1422025


git-svn-id: https://svn.apache.org/repos/asf/tiles/framework/branches/TILES_3_0_X@1484309 13f79535-47bb-0310-9956-ffa450edef68

michaelsembwever committed:
add link to Download page on each microsite


git-svn-id: https://svn.apache.org/repos/asf/tiles/framework/branches/TILES_3_0_X@1486560 13f79535-47bb-0310-9956-ffa450edef68

michaelsembwever committed:
compile against Java6


git-svn-id: https://svn.apache.org/repos/asf/tiles/framework/branches/TILES_3_0_X@1486577 13f79535-47bb-0310-9956-ffa450edef68

michaelsembwever committed:
upgrade to tiles-master-5.
upgrade to svnpubsub system using maven-scm-publish-plugin.
update the release and building pages for devs (svnpubsub, and referencing existing apache docs).
Update release documentation: build requires Java7, and build requires maven-3.0.5+


git-svn-id: https://svn.apache.org/repos/asf/tiles/framework/branches/TILES_3_0_X@1536457 13f79535-47bb-0310-9956-ffa450edef68

michaelsembwever committed:
Use the new tiles-master-6
 http://thread.gmane.org/gmane.comp.apache.tiles.devel/595


git-svn-id: https://svn.apache.org/repos/asf/tiles/framework/branches/TILES_3_0_X@1538333 13f79535-47bb-0310-9956-ffa450edef68

michaelsembwever committed:
[maven-release-plugin] prepare release tiles-parent-3.0.2

git-svn-id: https://svn.apache.org/repos/asf/tiles/framework/branches/TILES_3_0_X@1538337 13f79535-47bb-0310-9956-ffa450edef68

michaelsembwever committed:
[maven-release-plugin] prepare for next development iteration

git-svn-id: https://svn.apache.org/repos/asf/tiles/framework/branches/TILES_3_0_X@1538339 13f79535-47bb-0310-9956-ffa450edef68

michaelsembwever committed:
lessons learnt from the tiles-3.0.2 release


git-svn-id: https://svn.apache.org/repos/asf/tiles/framework/branches/TILES_3_0_X@1538384 13f79535-47bb-0310-9956-ffa450edef68

michaelsembwever committed:
TILES-573 – Tiles.xml definitions not reloaded when using expressions
Contribution from Eric B.


git-svn-id: https://svn.apache.org/repos/asf/tiles/framework/branches/TILES_3_0_X@1539385 13f79535-47bb-0310-9956-ffa450edef68

michaelsembwever committed:
[maven-release-plugin] prepare release tiles-parent-3.0.3

git-svn-id: https://svn.apache.org/repos/asf/tiles/framework/branches/TILES_3_0_X@1539477 13f79535-47bb-0310-9956-ffa450edef68

michaelsembwever committed:
[maven-release-plugin] prepare for next development iteration

git-svn-id: https://svn.apache.org/repos/asf/tiles/framework/branches/TILES_3_0_X@1539480 13f79535-47bb-0310-9956-ffa450edef68

michaelsembwever committed:
fix documentation after feedback from Emi Lu.
Reference: http://article.gmane.org/gmane.comp.apache.tiles.user/661


git-svn-id: https://svn.apache.org/repos/asf/tiles/framework/branches/TILES_3_0_X@1539840 13f79535-47bb-0310-9956-ffa450edef68

michaelsembwever committed:
-Duser.name needs to be defined inside the deploy goal too


git-svn-id: https://svn.apache.org/repos/asf/tiles/framework/branches/TILES_3_0_X@1539841 13f79535-47bb-0310-9956-ffa450edef68

michaelsembwever committed:
add mustache to the list of templating languages supported


git-svn-id: https://svn.apache.org/repos/asf/tiles/framework/branches/TILES_3_0_X@1539842 13f79535-47bb-0310-9956-ffa450edef68

michaelsembwever committed:
new website style, and update after tiles-3.0.3 release.
proper trademark and logo attribute.

references: 
 - http://thread.gmane.org/gmane.comp.apache.tiles.devel/602
 - http://svn.apache.org/viewvc?view=revision&revision=1539491
 - http://thread.gmane.org/gmane.comp.apache.tiles.devel/599
 - http://www.apache.org/foundation/marks/pmcs#graphics


git-svn-id: https://svn.apache.org/repos/asf/tiles/framework/branches/TILES_3_0_X@1541483 13f79535-47bb-0310-9956-ffa450edef68

michaelsembwever committed:
redo r1541483
 fixing links in site.xml ( http://jira.codehaus.org/browse/MSITE-159 )


git-svn-id: https://svn.apache.org/repos/asf/tiles/framework/branches/TILES_3_0_X@1541580 13f79535-47bb-0310-9956-ffa450edef68

michaelsembwever committed:
TILES-574 – Tiles expressions do not work after wildcard


git-svn-id: https://svn.apache.org/repos/asf/tiles/framework/branches/TILES_3_0_X@1543135 13f79535-47bb-0310-9956-ffa450edef68

michaelsembwever committed:
TILES-576 – Please delete old releases from mirroring system


git-svn-id: https://svn.apache.org/repos/asf/tiles/framework/branches/TILES_3_0_X@1546198 13f79535-47bb-0310-9956-ffa450edef68

michaelsembwever committed:
Use blank strings instead of nulls as placeholder values when replacing placeholders.

 Reference from mailing list:
  http://thread.gmane.org/gmane.comp.apache.tiles.user/685
     
>  <definition name="layout" template="/WEB-INF/layout/web.jsp"/>
>   <definition name="layout.mob" template="/WEB-INF/layout/mob.jsp"/>
>   <definition name="layout.tab" template="/WEB-INF/layout/tab.jsp"/>
>   <definition name="REGEXP:home(\.(mob|tab))?" extends="layout{1}">
>     <put-attribute name="heading" value="Home" />
>     <put-list-attribute name="body">
>       <add-attribute value="/WEB-INF/views/home.jsp" />
>     </put-list-attribute>
>   </definition>
>
> The above rules match a pattern employed with spring-mobile where
>  views can be resolved based on (mob)ile/(tab)let/normal , as normal is
>  a fallback it does not contain a suffix.
> 
> I can understand why this fails as the capture {1} does not exist in
>  the found pattern, instead I'd like it to recognise this and evaluate
>  to an empty string.
> 
> Full stack-trace is as follows: 
> java.lang.NullPointerException
>        at org.apache.tiles.BasicAttributeContext.inherit(BasicAttributeContext.java:212)
>        at org.apache.tiles.definition.dao.ResolvingLocaleUrlDefinitionDAO.getDefinitionFromResolver(ResolvingLocaleUrlDefinitionDAO.java:84)
>        …
      

The code ends up looking for "layoutnull" instead of "layout".

This comes from in RegexpDefinitionPatternMatcher.createDefinition(..) the call to matcher.group(i) returning null.

Rather than fix it there though i'd rather put the fix in lower down in PatternUtil.replacePlaceholders(..) so that any nulls in the vars parameter are always replaced with blanks.


git-svn-id: https://svn.apache.org/repos/asf/tiles/framework/branches/TILES_3_0_X@1571643 13f79535-47bb-0310-9956-ffa450edef68

michaelsembwever committed:
TILES-571 – Please make the significance of underscore (_) in tiles definition filename more obvious
SPR-11491 – Doc: Tiles 3 and underscores in definition names

<quote> 
… it looks like there is an issue with PostfixedApplicationResource.
 
This issue comes from SpringWildcardServletTilesApplicationContext#getResources(String) -> URLApplicationResource(String, URL) constructor -> super PostfixedApplicationResource(String localePath) constructor. When there is an
underscore, the string after the last underscore is identified as the locale. But clearly it's not a locale and one should be able to load /WEB-INF/base_tiles.xml.
</quote>

Along with improving the documentation the following code changes have been made
• if the resource name contains "_" and after it there is no supported locale language code then the whole resource name is treated as a non-localized path.
        This situation prints a warning like 
      		> No supported matching language for locale "zz". Using /my/path_zz.html as a non-localized resource path.
• if the resource name contains "_" and after it there is a supported locale language code but the locale constructed isn't valid/supported then the closest supported locale is used. This situation prints a warning like         > 
For resource /my/path_en_ZZ.html the closest supported matching locale to "en_ZZ" is "en". Using /my/path.html as resource path. 


git-svn-id: https://svn.apache.org/repos/asf/tiles/framework/branches/TILES_3_0_X@1583141 13f79535-47bb-0310-9956-ffa450edef68

michaelsembwever committed:
clean up slf4j dependencies
 - tiles should declare slf4j-api as depedency to be included transitively since its code references it,
 - tiles should not declare at all any slf4j implementation dependency in compile or runtime scope,
 - include in all test scopes sl4j-simple
 - bump to slf4j 1.7.6

reference: http://www.slf4j.org/faq.html#maven2


git-svn-id: https://svn.apache.org/repos/asf/tiles/framework/branches/TILES_3_0_X@1583199 13f79535-47bb-0310-9956-ffa450edef68

michaelsembwever committed:
tiles-request-1.0.4


git-svn-id: https://svn.apache.org/repos/asf/tiles/framework/branches/TILES_3_0_X@1592279 13f79535-47bb-0310-9956-ffa450edef68

michaelsembwever committed:
[maven-release-plugin] prepare release tiles-parent-3.0.4

git-svn-id: https://svn.apache.org/repos/asf/tiles/framework/branches/TILES_3_0_X@1592281 13f79535-47bb-0310-9956-ffa450edef68

michaelsembwever committed:
[maven-release-plugin] prepare for next development iteration

git-svn-id: https://svn.apache.org/repos/asf/tiles/framework/branches/TILES_3_0_X@1592283 13f79535-47bb-0310-9956-ffa450edef68

michaelsembwever committed:
further fixes on 
     TILES-574 – Tiles expressions do not work after wildcard

PatternUtil.INVALID_FORMAT_ELEMENT, the pattern to match "{.*}" occurrances that are not meant to be replaced, was in some cases was matching two much.
 see test PatternUtilTest.testReplacePlaceholdersEL_options
This resulted in "PatternSyntaxException: named capturing group is missing trailing }"

3 tests have been added to cover such various side-cases.


git-svn-id: https://svn.apache.org/repos/asf/tiles/framework/branches/TILES_3_0_X@1594481 13f79535-47bb-0310-9956-ffa450edef68

michaelsembwever committed:
tiles-request-1.0.5-SNAPSHOT


git-svn-id: https://svn.apache.org/repos/asf/tiles/framework/branches/TILES_3_0_X@1598398 13f79535-47bb-0310-9956-ffa450edef68

nlebas committed:
TILES-580 getAsString does not support expressions

git-svn-id: https://svn.apache.org/repos/asf/tiles/framework/branches/TILES_3_0_X@1625204 13f79535-47bb-0310-9956-ffa450edef68

nlebas committed:
fix jenkins

git-svn-id: https://svn.apache.org/repos/asf/tiles/framework/branches/TILES_3_0_X@1625208 13f79535-47bb-0310-9956-ffa450edef68

nlebas committed:
bump to tiles-request 1.0.6

git-svn-id: https://svn.apache.org/repos/asf/tiles/framework/branches/TILES_3_0_X@1626663 13f79535-47bb-0310-9956-ffa450edef68

nlebas committed:
[maven-release-plugin] prepare release tiles-parent-3.0.5

git-svn-id: https://svn.apache.org/repos/asf/tiles/framework/branches/TILES_3_0_X@1626664 13f79535-47bb-0310-9956-ffa450edef68

nlebas committed:
[maven-release-plugin] prepare for next development iteration

git-svn-id: https://svn.apache.org/repos/asf/tiles/framework/branches/TILES_3_0_X@1626666 13f79535-47bb-0310-9956-ffa450edef68

michaelsembwever committed:
TILES-583 changed the broken link

Just changed the link in the web framework integrations page to point to the version of spring-mvc that has the sample code in it's repository.

Contribution from Clare Donnelly
 ref: https://github.com/apache/tiles/pull/3


git-svn-id: https://svn.apache.org/repos/asf/tiles/framework/branches/TILES_3_0_X@1666175 13f79535-47bb-0310-9956-ffa450edef68

nlebas committed:
TILES-584: EL expressions not working for list attributes

git-svn-id: https://svn.apache.org/repos/asf/tiles/framework/branches/TILES_3_0_X@1666705 13f79535-47bb-0310-9956-ffa450edef68

michaelsembwever committed:
Documentation spelling errors fixed – https://github.com/apache/tiles/pull/6
contributed by Tim Gifford ( https://github.com/timgifford )


git-svn-id: https://svn.apache.org/repos/asf/tiles/framework/branches/TILES_3_0_X@1706314 13f79535-47bb-0310-9956-ffa450edef68

michaelsembwever committed:
TILES-585 – Definition cascade attributes duplicated values under high load
Fix by Jakub Dziworski - jakub.dziworski@gmail.com
This closes #9
ref: https://github.com/apache/tiles/pull/9/commits/78d9001ea5d0b02523ada36453848125d80ec6e3


git-svn-id: https://svn.apache.org/repos/asf/tiles/framework/branches/TILES_3_0_X@1752405 13f79535-47bb-0310-9956-ffa450edef68

michaelsembwever committed:
[maven-release-plugin] prepare release tiles-parent-3.0.6

git-svn-id: https://svn.apache.org/repos/asf/tiles/framework/branches/TILES_3_0_X@1752631 13f79535-47bb-0310-9956-ffa450edef68

michaelsembwever committed:
[maven-release-plugin] prepare for next development iteration

git-svn-id: https://svn.apache.org/repos/asf/tiles/framework/branches/TILES_3_0_X@1752633 13f79535-47bb-0310-9956-ffa450edef68

michaelsembwever committed:
use Tiles AutoTag-1.2


git-svn-id: https://svn.apache.org/repos/asf/tiles/framework/branches/TILES_3_0_X@1755952 13f79535-47bb-0310-9956-ffa450edef68

michaelsembwever committed:
[maven-release-plugin] prepare release tiles-parent-3.0.7

git-svn-id: https://svn.apache.org/repos/asf/tiles/framework/branches/TILES_3_0_X@1755965 13f79535-47bb-0310-9956-ffa450edef68

michaelsembwever committed:
[maven-release-plugin] prepare for next development iteration

git-svn-id: https://svn.apache.org/repos/asf/tiles/framework/branches/TILES_3_0_X@1755967 13f79535-47bb-0310-9956-ffa450edef68

michaelsembwever committed:
fix log and release docs (svnpub)


git-svn-id: https://svn.apache.org/repos/asf/tiles/framework/branches/TILES_3_0_X@1810179 13f79535-47bb-0310-9956-ffa450edef68

michaelsembwever committed:
use tiles-request-1.0.7


git-svn-id: https://svn.apache.org/repos/asf/tiles/framework/branches/TILES_3_0_X@1810188 13f79535-47bb-0310-9956-ffa450edef68

michaelsembwever committed:
[maven-release-plugin] prepare release tiles-parent-3.0.8

git-svn-id: https://svn.apache.org/repos/asf/tiles/framework/branches/TILES_3_0_X@1810193 13f79535-47bb-0310-9956-ffa450edef68

michaelsembwever committed:
[maven-release-plugin]  copy for tag tiles-parent-3.0.8

git-svn-id: https://svn.apache.org/repos/asf/tiles/framework/tags/tiles-parent-3.0.8@1810194 13f79535-47bb-0310-9956-ffa450edef68

michaelsembwever committed:
[maven-release-plugin] prepare for next development iteration

git-svn-id: https://svn.apache.org/repos/asf/tiles/framework/branches/TILES_3_0_X@1810195 13f79535-47bb-0310-9956-ffa450edef68

michaelsembwever committed:
fix generation of digests and signatures in assembly subproject


git-svn-id: https://svn.apache.org/repos/asf/tiles/framework/tags/tiles-parent-3.0.8@1814482 13f79535-47bb-0310-9956-ffa450edef68

michaelsembwever committed:
svn merge -c1814482 ^/tiles/framework/tags/tiles-parent-3.0.8


git-svn-id: https://svn.apache.org/repos/asf/tiles/framework/branches/TILES_3_0_X@1814484 13f79535-47bb-0310-9956-ffa450edef68

nlebas commented:
Security roles in that context are the standard JavaEE concept. See the documentation for your application server, for instance:
- https://docs.oracle.com/cd/E19798-01/821-1841/bncav/index.html
- https://tomcat.apache.org/tomcat-7.0-doc/realm-howto.html
- http://www.eclipse.org/jetty/documentation/9.4.x/configuring-security.html
- https://docs.jboss.org/author/display/WFLY10/Security+Realms
- ...

nlebas commented:
Please close the pull request as it would be inappropriate to merge TILES_3_0_X into trunk.

michaelsembwever committed:
Backport r1840285 
svn merge -c1840285 ^/tiles/framework/trunk


git-svn-id: https://svn.apache.org/repos/asf/tiles/framework/branches/TILES_3_0_X@1840286 13f79535-47bb-0310-9956-ffa450edef68

michaelsembwever committed:
fix build – bump maven-remote-resources-plugin to 1.5


git-svn-id: https://svn.apache.org/repos/asf/tiles/framework/branches/TILES_3_0_X@1840289 13f79535-47bb-0310-9956-ffa450edef68

michaelsembwever committed:
Fix stuck thread on hashmap
https://issues.apache.org/jira/browse/TILES-599
Patch by shakeelabbas1
This closes #16


git-svn-id: https://svn.apache.org/repos/asf/tiles/framework/branches/TILES_3_0_X@1840290 13f79535-47bb-0310-9956-ffa450edef68
